src/Features/Core/Portable/EditAndContinue/EditSession.cs

```diff
@@ -342,16 +341,62 @@ private static async Task PopulateChangedAndAddedDocumentsAsync(CommittedSolutio
                 changedOrAddedDocuments.Add(document);
             }
 
-            foreach (var documentId in changes.GetAddedDocuments())
+            foreach (var documentId in newProject.State.DocumentStates.GetAddedStateIds(oldProject.State.DocumentStates))
             {
-                var document = project.GetDocument(documentId)!;
+                var document = newProject.GetDocument(documentId)!;
                 if (document.State.Attributes.DesignTimeOnly)
                 {
                     continue;
                 }
 
                 changedOrAddedDocuments.Add(document);
             }
+
+            // TODO: support document removal/rename (see https://github.com/dotnet/roslyn/issues/41144, https://github.com/dotnet/roslyn/issues/49013).
+
+            // Given the following assumptions:
+            // - source generators are deterministic,
+            // - source documents, metadata references and compilation options have not changed,
+            // - additional documents have not changed,
+            // - analyzer config documents have not changed,
+            // the outputs of source generators will not change.
```

Comment:
So what's the intent here for project-to-project references? That you're not running? I don't disagree with the decision, I wasn't sure if there were some other cases you have in mind.

Reply:
You mean the generator depending on content of other projects to which it gets thru references of the compilation?

Reply:
How does the IDE deal with that for other services, like completion?